### PR TITLE
feat: Support render props for <Popover/> and <Popconfirm/>

### PR DESCRIPTION
**Pull request by nossbigg**
# 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/21935

### 💡 Background and solution

Problem:
Current `<Popover/>` and `<Popconfirm/>` doesn't support render props.

Solution:
Add support for render props for `<Popover/>` and `<Popconfirm/>`

```javascript
<Popover content={() => 'some-content'} title={() => 'some-title'} />

<Popconfirm title={() => 'some-title'} />
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Add render props support for Popover/Popconfirm  |
| 🇨🇳 Chinese |  Add render props support for Popover/Popconfirm  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/popover/index.en-US.md](https://github.com/nossbigg/ant-design/blob/popover-renderprop/components/popover/index.en-US.md)
[View rendered components/popover/index.zh-CN.md](https://github.com/nossbigg/ant-design/blob/popover-renderprop/components/popover/index.zh-CN.md)

### CODE REVIEW

**Comment by nossbigg** [comment]
Mine's a realllllly old version 😅, rebasing...

File: components/popover/index.zh-CN.md [hunk 1]
```diff
@@ -15,10 +15,10 @@ title: Popover
...
 ## API
 
-| 参数    | 说明     | 类型              | 默认值 | 版本 |
-| ------- | -------- | ----------------- | ------ | ---- |
-| content | 卡片内容 | string\|ReactNode | 无     |      |
-| title   | 卡片标题 | string\|ReactNode | 无     |      |
+| 参数    | 说明     | 类型                               | 默认值 | 版本 |
+| ------- | -------- | ---------------------------------- | ------ | ---- |
+| content | 卡片内容 | string\|ReactNode\|() => ReactNode | 无     |      |
+| title   | 卡片标题 | string\|ReactNode\|() => ReactNode | 无     |      |
```
Comment (afc163) [inline components/popover/index.zh-CN.md]: Popconfirm 的文档和 ts 定义也可以更新掉。

Comment (nossbigg) [reply components/popover/index.zh-CN.md]: Where else needs updating? (sorry i'm not good at chinese 😅)

Comment (afc163) [reply components/popover/index.zh-CN.md]: Sorry, I means Popconfirm's document and tsd.

https://ant.design/components/popconfirm/

Comment (yoyo837) [reply components/popover/index.zh-CN.md]: Please update its docs and ts definitions together.

Comment (nossbigg) [reply components/popover/index.zh-CN.md]: I did not introduce any changes to Popconfirm though 🤔

Comment (afc163) [reply components/popover/index.zh-CN.md]: Popconfirm has similar API with Popover, we can update them together.

Comment (nossbigg) [reply components/popover/index.zh-CN.md]: Updated Popconfirm, please review 👍

File: components/popconfirm/index.en-US.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ The difference with the `confirm` modal dialog is that it's more lightweight tha
...
-| okText | text of the Confirm button | string | `OK` |
+| cancelText | text of the Cancel button | string\|ReactNode\|() => ReactNode | `Cancel` |
+| okText | text of the Confirm button | string\|ReactNode\|() => ReactNode | `OK` |
 | okType | Button `type` of the Confirm button | string | `primary` |
-| title | title of the confirmation box | string\|ReactNode | - |
+| title | title of the confirmation box | string\|ReactNode\|() => ReactNode | - |
 | onCancel | callback of cancel | function(e) | - |
 | onConfirm | callback of confirmation | function(e) | - |
-| icon | customize icon of confirmation | ReactNode | `<ExclamationCircle />` |
+| icon | customize icon of confirmation | ReactNode\|() => ReactNode | `<ExclamationCircle />` |
```
Comment (afc163) [inline components/popconfirm/index.en-US.md]: `title` is enough. I perfer to keep `okText`, `cancelText` and `icon` as ReactNode unless more requirements coming.

Comment (nossbigg) [reply components/popconfirm/index.en-US.md]: Sure, will make the amendments 👍

Comment (nossbigg) [reply components/popconfirm/index.en-US.md]: Done

**Review by afc163 — APPROVED** [review]

**Comment by nossbigg** [comment]
@afc163 Thanks for the merge! 🙂

**Pull request merged**